ophidian/src/ophidian/finder.py

```diff
@@ -0,0 +1,132 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import os
+import re
+import sys
+import glob
+import json
+import subprocess
+
+import attr
+
+
+VALID_PYTHON_NAMES = re.compile(r"""
+    ^
+    (python|pypy)  # We currently only support these two implementations
+    (-?(\d\.?)+)?
+    (\.exe)?
```

Comment:
Yeah, this is somewhat aspirational.

Right now it would be nice if the discovery code worked on Windows but it probably doesn't, the launcher code obviously doesn't and neither does the installer (pyenv doesn't work at all on Windows.

Long run I definitely want this to grow Windows support. My plan is to write a second installer implementation, probably based on anaconda), and a powershell script as a launcher, but I don't plan to do that as part of this PR.